lib/build-head.js

```diff
@@ -1,51 +1,48 @@
 'use strict';
 
-var fetchFiles = require('./common').fetchFiles;
+const Joi = require('joi');
+const Dynastar = require('dynastar');
 
 /**
- * function buildHead(datastar)
- * Returns a prototypal datastar model representing
+ * function buildHead(dynamo)
+ * Returns a prototypal dynamo model representing
  * HEAD of a built package being persisted to the connection
- * associated with the `datastar` instance provided.
+ * associated with the `dynamo` instance provided.
  *
- * @param {datastar} datastar Instance of datastar to attach to.
+ * @param {dynamo} dynamo Instance of dynamo to attach to.
  * @param {Object} models Set of registry-data models.
  *
- * @returns {Object} an datastar model representing the metadata of
+ * @returns {Object} an dynamo model representing the metadata of
  * the latest build (i.e. HEAD) of a [env, package] pair
  */
-module.exports = function buildHead(datastar, models) {
-  var cql = datastar.schema.cql;
+module.exports = function buildHead(dynamo) {
+  const createKey = (data) => {
+    return [data.name, data.env, data.version].join('!');
+  };
+  const rangeKey = 'locale';
 
-  var BuildHead = datastar.define('build_head', {
-    schema: datastar.schema.object({
-      env: cql.text(),
-      name: cql.text(),
-      build_id: cql.text(),
-      previous_build_id: cql.text(),
-      rollback_build_ids: cql.map(cql.text(), cql.text()), // timestamp is stored as string
-      create_date: cql.timestamp({ default: 'create' }),
-      udpate_date: cql.timestamp({ default: 'update' }),
-      version: cql.text(),
-      locale: cql.text(),
-      cdn_url: cql.text(), // URL of CDN to be used as a base for all the artifacts
-      fingerprints: cql.set(cql.text()),
-      artifacts: cql.set(cql.text()),
-      recommended: cql.set(cql.text())
-    }).partitionKey(['env', 'name'])
-      .clusteringKey('locale'),
-    with: {
-      compaction: {
-        class: 'LeveledCompactionStrategy'
-      }
+  const model = dynamo.define('BuildHead', {
+    hashKey: 'key',
+    rangeKey,
+    timestamps: true, // adds updatedAt and createdAt
+    tableName: 'WarehouseBuildHead',
+    schema: {
+      key: Joi.string(),
+      env: Joi.string(),
+      name: Joi.string(),
+      buildId: Joi.string(),
+      previousBuildId: Joi.string(),
```

Comment:
we might want to allow(null) here in case it appears when migrating but not sure it makes a difference
```suggestion
      previousBuildId: Joi.string().allow(null),
```